### PR TITLE
ISPN-9944 ServerNG security + Test framework

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-9944

What works (checked items have tests):

- [x] Single Port with and without TLS
- [x] SASL authentication 
  - [x] PLAIN
  - [x] DIGEST-*
  - [x] SCRAM-SHA-*
  - [x] EXTERNAL
  - [ ] GSSAPI
  - [ ] GS2
  - [ ] OAuth2
- [x] Realms
  - [x] Properties (the old user/roles.properties)
  - [x] LDAP
  - [x] TrustStore
  - [ ] Filesystem (parser done, not tested)
  - [ ] Kerberos (parser done, not tested)
  - [ ] Token (parser done, not tested)
- [x] Elytron integrated into the client
- [x] Test framework
  - [x] Server ClassRule for starting/stopping servers for a suite of tests
  - [x] Server Rule for creating per-test caches and removing them
  - [x] Embedded server driver
  - [x] LDAP Rule for starting/stopping the directory server
  - [ ] Container server driver (based on test-containers, incomplete)
  - [ ] Forked server driver
- [ ] XSD (incomplete)

### CODE REVIEW

File: server/runtime/pom.xml [hunk 1]
```diff
@@ -89,6 +89,12 @@
          <artifactId>httpclient</artifactId>
          <scope>test</scope>
       </dependency>
+      <dependency>
```
Comment (karesti) [inline server/runtime/pom.xml]: 👍

**Comment by danberindei** [comment]
Might be related? https://ci.infinispan.org/job/Infinispan/job/PR-7044/1/testReport/junit/org.infinispan.tasks/TaskManagerTest/testRegisterDuplicateEngine/

File: server/runtime/src/test/java/org/infinispan/server/functional/ClusteredTests.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 10.0
+ **/
+@RunWith(Suite.class)
+@Suite.SuiteClasses({CacheOperations.class, CounterOperations.class})
+public class ClusteredTests {
+
+   @ClassRule
+   public static final InfinispanServerRule SERVERS = new InfinispanServerRule(new InfinispanServerTestConfiguration("configuration/ClusteredServerTest.xml").numServers(2));
```
Comment (gustavonalle) [inline server/runtime/src/test/java/org/infinispan/server/functional/ClusteredTests.java]: Pure awesomeness :)

File: tasks/manager/src/main/java/org/infinispan/tasks/logging/Log.java [hunk 3]
```diff
@@ -13,8 +13,8 @@
  */
 @MessageLogger(projectCode = "ISPN")
 public interface Log extends BasicLogger {
-   @Message(value = "Task Engine '%s' has already been registered", id = 27001)
-   IllegalStateException duplicateTaskEngineRegistration(String taskEngineName);
+   /*@Message(value = "Task Engine '%s' has already been registered", id = 27001)
+   IllegalStateException duplicateTaskEngineRegistration(String taskEngineName);*/
```
Comment (gustavonalle) [inline tasks/manager/src/main/java/org/infinispan/tasks/logging/Log.java]: remove?

**Comment by gustavonalle** [comment]
LGTM, just a trivial cleanup that can be done later, pulling...

**Comment by gustavonalle** [comment]
Integrated, thanks @tristantarrant

**Pull request merged**